File: 21주차/피로도.py
from itertools import permutations
def solution(k,d):
    answer = 0
    arr = [i for i in range(len(d))]
    for i in list(permutations(arr,len(d))):
        cnt = 0
        tmp = k
        for step in i:
            if tmp >= d[step][0]:
                tmp -= d[step][1]
                cnt += 1
        answer = max(answer,cnt)
    
    return answer

File: 47주차/피로도.js
function solution(k, dungeons) {
  var answer = -1;

  let n = dungeons.length;
  let visited = new Array(n).fill(0);

  function dfs(k, cnt) {
    //현재 남은 피로도, 최대 개수.
    answer = Math.max(answer, cnt);

    for (let i = 0; i < n; i++) {
      if (!visited[i] && k >= dungeons[i][0]) {
        // 남은 피로도가 필요필요도 이상이고, 방문 안했다면,
        visited[i] = 1;
        dfs(k - dungeons[i][1], cnt + 1);
        visited[i] = 0;
      }
    }
  }
  dfs(k, 0);

  return answer;
} // 프로그래머스 완전탐색 문제
